sources/api/schnauzer/src/helpers.rs

```diff
@@ -463,6 +479,43 @@ pub fn ecr_prefix(
     Ok(())
 }
 
+/// `host` takes an absolute URL and trims it down and returns its host.
+pub fn host(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    trace!("Starting host and port helper");
+    let template_name = template_name(renderctx);
+    check_param_count(helper, template_name, 1)?;
+
+    let url_val = get_param(helper, 0)?;
+    let url_str = url_val
+        .as_str()
+        .with_context(|| error::InvalidTemplateValue {
+            expected: "string",
+            value: url_val.to_owned(),
+            template: template_name.to_owned(),
+        })?;
+    let url = Url::parse(url_str).with_context(|| error::UrlParse {
+        value: url_val.to_owned(),
+        template: template_name,
+    })?;
+    let url_host = url.host_str().context(error::UrlHost {
+        url: url.to_owned(),
```

Comment:
I have to clone here because `.host_str()` borrows `url`

Reply:
Normally Snafu handles ownership/cloning of referenced objects, though.  You can pass `&str` in `context` with `String` in the error type, for the most common example.  Is that not working here?  (I'm not sure of the exact type bound required.)

Reply:
When I try that I get the following, note that `url` is a `url::Url`:
```
error[E0505]: cannot move out of `url` because it is borrowed
   --> api/schnauzer/src/helpers.rs:506:48
    |
506 |     let url_host = url.host_str().with_context(|| error::UrlHost {
    |                    ---            ------------ ^^ move out of `url` occurs here
    |                    |              |
    |                    |              borrow later used by call
    |                    borrow of `url` occurs here
507 |         url: url,
    |              --- move occurs due to use in closure

```

Reply:
I guess I can just make it take a String, and just pass `url_str`